src/sentry/models/release.py

```diff
@@ -444,6 +448,19 @@ def set_commits(self, commit_list):
                     last_commit_id=latest_commit.id if latest_commit else None,
                 )
 
+        # fill any missing ReleaseHeadCommit entries
+        for repo_id, commit_id in six.iteritems(head_commit_by_repo):
+            try:
+                with transaction.atomic():
+                    ReleaseHeadCommit.objects.create(
+                        organization_id=self.organization_id,
+                        release=self,
+                        repository_id=repo_id,
+                        commit=commit_id,
+                    )
+            except IntegrityError:
```

Comment:
should only really error when ``set_refs`` was done -- it might be worth us just overwriting refs though?